### PR TITLE
Approve staking transfer, added GenesisProtocol.getExecutedDaoProposals

### PR DESCRIPTION
**Pull request by dkent600**
Addresses:

https://github.com/daostack/arc.js/issues/125
https://github.com/daostack/arc.js/issues/119

Threw in `VestingScheme.getAgreements`.

Regarding approval token transfers:  The added approvals only happen when `Config.get("autoApproveTokenTransfers")` is true, which is the default (so client apps can choose to disable it).

The approvals are happening in:

`ContributionReward.proposeContributionReward`
`GenesisProtocol.stake`
`VestingScheme.create`

### CODE REVIEW

File: lib/contracts/contributionreward.ts [hunk 1]
```diff
@@ -118,6 +120,19 @@ export class ContributionRewardWrapper extends ExtendTruffleContract {
...
+    const schemeParams = await controller.getSchemeParameters(this.address, options.avatar);
+
+    const orgNativeTokenFee = (await this.contract.parameters(schemeParams))[0];
+    if (Config.get("autoApproveTokenTransfers") && (orgNativeTokenFee > 0)) {
+      /**
+       * approve immediate transfer of native tokens from msg.sender to the avatar
+       */
+      const avatarService = new AvatarService(options.avatar);
+      const token = await avatarService.getNativeToken();
+      await token.approve(this.address, orgNativeTokenFee, { from: Utils.getDefaultAccount() });
```
Comment (tibetsprague) [inline lib/contracts/contributionreward.ts]: why did this work before, without approval?

File: lib/contracts/genesisProtocol.ts [hunk 2]
```diff
@@ -791,11 +810,46 @@ export class GenesisProtocolWrapper extends ExtendTruffleContract {
     return state;
   }
 
-  public async _validateVote(vote: number, proposalId: Hash): Promise<void> {
-    const numChoices = await this.getNumberOfChoices({ proposalId });
-    if (!Number.isInteger(vote) || (vote < 0) || (vote > numChoices)) {
-      throw new Error("vote is not valid");
+  /**
+   * Return all GenesisProtocol proposals ever created under the given avatar.
```
Comment (tibetsprague) [inline lib/contracts/genesisProtocol.ts]: not created proposals, only executed

File: lib/contracts/contributionreward.ts (hunk 1 above)
Comment (dkent600) [reply lib/contracts/contributionreward.ts]: There were no tests for the case where orgNativeTokenFee was > 0.  Can't speak for Alchemy.

File: index.d.ts [hunk 3]
```diff
@@ -334,11 +334,15 @@ declare module "@daostack/arc.js" {
    * fired by voting machines
    */
   export interface ExecuteProposalEventResult {
-    _decision: number;
+    _decision: BigNumber.BigNumber;
```
[the remark this answers is not in the document]
Comment (dkent600) [reply index.d.ts]: In fact, _decision _is_ coming in as a BigNumber

File: lib/contracts/commonEventInterfaces.ts [hunk 4]
```diff
@@ -15,11 +15,15 @@ export interface NewProposalEventResult {
  * fired by voting machines
  */
 export interface ExecuteProposalEventResult {
-  _decision: number;
+  _decision: BigNumber.BigNumber;
```
[the remark this answers is not in the document]
Comment (dkent600) [reply lib/contracts/commonEventInterfaces.ts]: same here

File: index.d.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (tibetsprague) [reply index.d.ts]: ah, right you are, I'm just converting straight to a Number.

**Pull request merged**